packages/babel-preset-typescript/src/plugin-rewrite-ts-imports.ts

```diff
@@ -0,0 +1,22 @@
+import { declare } from "@babel/helper-plugin-utils";
+import type { types as t } from "@babel/core";
+import type { NodePath } from "@babel/traverse";
+
+export default declare(function () {
+  return {
+    visitor: {
+      "ImportDeclaration|ExportAllDeclaration|ExportNamedDeclaration"(
+        path: NodePath<
+          | t.ImportDeclaration
+          | t.ExportAllDeclaration
+          | t.ExportNamedDeclaration
+        >,
+      ) {
+        const { source } = path.node;
+        if (source) {
+          source.value = source.value.replace(/(\.[mc]?)ts$/, "$1js");
```

Comment:
```suggestion
          source.value = source.value.replace(/(\/.+?\.[mc]?)ts$/, "$1js");
```
To cover the edge case mentioned by @bakkot, unless the import is specified as `import "/tmp/soundcloud.ts"`, which we can't tackle without resolving it first.

Reply:
`if (!source.includes("@"))`

We may also want to exclude `@`, such as `@org/mod.ts`.

Reply:
I would only special-case import specifiers that contain no `\` or `/`, because the only way a specifier without `/` could work is if it maps to a folder (using Node.js' resolution algorithm) or to a file without any rewriting (using an import map).

For example, I have a project with an import map that includes the following:
```json
{
  "imports": {
    "@/": "./",
    "@components/": "./frontend/components/"
```

As such, `@/foo.ts` and `@components/foo.ts` both map to files and not to directories (and thus they would need their `.ts` extension to be rewritten).

I would prefer to either (in other of preference):
1. do nothing, and if somebody has some imports that must not be rewritten they can disable the option for that file
2. only ignore imports with no `/`
3. extract the plugin to its own package, and when it's used directly allow passing a function `originalSpecifier => transformedSpecifier` to allow users to filter and transform however they want

However, before doing (3) I would wait for someone to open an issue with a concrete use case.

Reply:
> extract the plugin to its own package, and when it's used directly allow passing a function originalSpecifier => transformedSpecifier to allow users to filter and transform however they want

We can even allow `rewriteImportExtensions` to pass in a function in the future.😉